### PR TITLE
Report error exit code when any error is reported

### PR DESCRIPTION
**Pull request by marek-safar**
also don't produce any output binaries when error occurs

### CODE REVIEW

File: src/linker/Linker.Steps/OutputStep.cs [hunk 1]
```diff
@@ -76,11 +76,15 @@ static TargetArchitecture CalculateArchitecture (TargetArchitecture readyToRunAr
 			throw new BadImageFormatException ("unrecognized module attributes");
 		}
 
+		protected override bool ConditionToProcess ()
```
Comment (mateoatr) [inline src/linker/Linker.Steps/OutputStep.cs:79]: I think after ProcessWarningsStep, there's no step that could produce any warning/error. Should these steps have this condition as well, or maybe short-circuit steps after this point if errors were found?

File: src/linker/Linker/Driver.cs [hunk 2]
```diff
@@ -725,6 +721,7 @@ protected int SetupContext (ILogger customLogger = null)
 			// MarkStep
 			// ReflectionBlockedStep [optional]
 			// ProcessWarningsStep
+			// OutputWarningSuppressions
```
Comment (mateoatr) [inline src/linker/Linker/Driver.cs:724]: nit: I think this doesn't match its actual place in the pipeline.

Comment (marek-safar) [reply src/linker/Linker/Driver.cs:724]: yeah, forgot to update the code. The comment is correct.

File: src/linker/Linker.Steps/OutputStep.cs (hunk 1 above)
Comment (marek-safar) [reply src/linker/Linker.Steps/OutputStep.cs:79]: I don't think it's worth it (optimize for failure), they should not produce errors but they could in the future.

**Review by vitek-karas — APPROVED** [review]

**Pull request merged**